src/main/java/seedu/duke/Parser.java

```diff
@@ -0,0 +1,117 @@
+package seedu.duke;
+
+import java.io.IOException;
+import java.time.LocalDate;
+
+/**
+ * Represents a Parser, makes sense of user input
+ */
+public class Parser {
+
+    private TaskList tasks;
+    private Ui ui;
+    private Storage storage;
+
+    /**
+     * Instantiates a new Parser object
+     */
+    public Parser(TaskList taskL, Ui ui, Storage storage) {
+        this.tasks = taskL;
+        this.ui = ui;
+        this.storage = storage;
+    }
+    public static class DukeException extends Exception {
+        public DukeException(String msg) {
+            super(msg);
+        }
+    }
+    public static class EmptyMessageException extends DukeException {
+        public EmptyMessageException() {
+            super("☹ OOPS!!! The description of a todo cannot be empty.");
+        }
+    }
+    public static class InvalidCommandException extends DukeException {
+        public InvalidCommandException() {
+            super("☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
+        }
+    }
+
+    private final String[] commandWords = new String[]{"list", "mark",
+            "unmark", "todo", "event", "deadline", "delete", "bye", "find"};
+
+    public void parse(String text) throws EmptyMessageException, InvalidCommandException {
```

Comment:
Should all public methods have Javadoc comments except getters/setters? I noticed the same issue in several places too.